src/pluginlist.cpp

```diff
@@ -182,82 +182,81 @@ void PluginList::refresh(const QString& profileName,
       gamePlugins ? gamePlugins->lightPluginsAreSupported() : false;
   const bool overridePluginsAreSupported =
       gamePlugins ? gamePlugins->overridePluginsAreSupported() : false;
+  const bool loadOrderMechanismNone =
+      m_GamePlugin->loadOrderMechanism() == IPluginGame::LoadOrderMechanism::None;
 
   m_CurrentProfile = profileName;
 
-  QStringList availablePlugins;
+  std::unordered_map<QString, FileEntryPtr> availablePlugins;
+  QStringList archiveCandidates;
 
-  std::vector<FileEntryPtr> files = baseDirectory.getFiles();
-  for (FileEntryPtr current : files) {
+  for (FileEntryPtr current : baseDirectory.getFiles()) {
     if (current.get() == nullptr) {
       continue;
     }
-    QString filename = ToQString(current->getName());
+    const QString& filename = ToQString(current->getName());
```

Comment:
It's not important, but for future reference, the change to this line does nothing. `ToQString` returns a `QString`, not a `const QString&`, so a `QString` has to be constructed in this stack frame anyway. There's no copy happening in the old or new version due to copy elision.